spring-cloud-stream-binders/spring-cloud-stream-binder-redis/src/test/java/org/springframework/cloud/stream/binder/redis/RedisBinderTests.java

```diff
@@ -128,14 +138,14 @@ public void testProducerProperties() throws Exception {
 		producerBinding.unbind();
 		assertFalse(producerEndpoint.isRunning());
 
-		Properties properties = new Properties();
-		properties.put("partitionKeyExpression", "'foo'");
-		properties.put("partitionKeyExtractorClass", "foo");
-		properties.put("partitionSelectorExpression", "0");
-		properties.put("partitionSelectorClass", "foo");
-		properties.put(BinderPropertyKeys.NEXT_MODULE_COUNT, "1");
+		ProducerProperties producerProperties = new ProducerProperties();
+		producerProperties.setPartitionKeyExpression("'foo'");
```

Comment:
these properties should really be mutually exclusive (validated within the props?):
-  `partitionKeyExpression` OR `partitionKeyExtractorClass`
-  `partitionSelectorExpression` OR `partitionSelectorClass`